auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -127,6 +126,10 @@ private void populateIdpList(List<IdpConfig> providers) {
                     loginButton = getLayoutInflater()
                             .inflate(R.layout.idp_button_twitter, btnHolder, false);
                     break;
+                case EmailAuthProvider.PROVIDER_ID:
+                    loginButton = getLayoutInflater()
+                            .inflate(R.layout.provider_button_email, btnHolder, false);
```

Comment:
Could we take this refactor a bit further?  `Provider` could define `@LayoutRes int getButtonLayout()` and then this switch could disappear and just become `loginButton = getLayoutInflater().inflate(provider.getButtonLayout(), btnHolder, false)`